File: Front/WebShop-Frontend/src/app/models/product.ts
import { Category } from "./category";
import { Image } from "./image";
import { ProductAttribute } from "./product-attribute";
import { User } from "./user";
import { Comment } from "./comment";

export class Product {
    id: number;
    name: string;
    description: string;
    price: number;
    isNew: boolean;
    location: string;
    creationDate : Date;
    quantity: number;
    status: ProductStatus;
    //sallerId: number;
    saller: User;
    images: Image[] = [];
    productAttributes: ProductAttribute[] = [];
    buyerId: number;
    categoryId: number;
    comments: Comment[] = [];
    //category: Category;

    constructor(id:number, name: string, description: string, price: number, isNew: boolean, location: string, creationDate: Date, quantity: number, status: ProductStatus, saller: User, images: Image[], productAttributes: ProductAttribute[], buyerId: number, categoryId: number, comments: Comment[]) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.price = price;
        this.isNew = isNew;
        this.location = location;
        this.creationDate = creationDate;
        this.quantity = quantity;
        this.status = status;
        this.saller = saller;
        this.images = images;
        this.buyerId = buyerId;
        //this.images = images;
        this.productAttributes = productAttributes;
        this.categoryId = categoryId;
        this.comments = comments;
    }
}

enum ProductStatus {
    ACTIVE,
    SOLD,
    INACTIVE
}

File: Back/WebShop-Backend/src/main/java/com/example/WebShop/models/dto/Product.java
package com.example.WebShop.models.dto;

import com.example.WebShop.models.enums.ProductStatus;
import jakarta.persistence.Basic;
import jakarta.persistence.Column;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.Data;

import java.sql.Timestamp;
import java.util.List;

@Data
public class Product {
    private Integer id;
    private String name;
    private String description;
    private Integer price;
    private boolean isNew;
    private String location;
    private Timestamp createDate;
    private Integer quantity;
    private ProductStatus status;
    //private Integer sallerId;
    private User saller;
    private Integer buyerId;
    private Integer categoryId;
    //private Category category;
    private List<Image> images;
    private List<ProductAttribute> productAttributes;
    private  List<Comment> comments;
}
